src/frontend/src/session/cursor_manager.rs

```diff
@@ -620,81 +751,53 @@ impl SubscriptionCursor {
         rw_timestamp: Option<u64>,
         dependent_table_id: &TableId,
         handler_args: HandlerArgs,
+        seek_pk_row: Option<Row>,
     ) -> Result<BatchQueryPlanResult> {
         let session = handler_args.clone().session;
         let table_catalog = session.get_table_by_id(dependent_table_id)?;
-        let pks = table_catalog.pk();
         let context = OptimizerContext::from_handler_args(handler_args.clone());
-        if let Some(rw_timestamp) = rw_timestamp {
-            let version_id = {
-                let version = session.env.hummock_snapshot_manager.acquire();
-                let version = version.version();
-                if !version
-                    .state_table_info
-                    .info()
-                    .contains_key(dependent_table_id)
-                {
-                    return Err(anyhow!("table id {dependent_table_id} has been dropped").into());
-                }
-                version.id
-            };
-            Self::create_batch_plan_for_cursor(
-                &table_catalog,
-                &session,
-                context.into(),
-                rw_timestamp,
-                rw_timestamp,
-                version_id,
-                pks,
-            )
-        } else {
```

Comment:
Previously we create a batch query by constructing a SQL when `rw_timestamp` is not specified. How do we handle it in this PR?

Reply:
Doing the same thing, just putting the logic into create_batch_plan_for_cursor.